### ISSUE TITLE
Example in the readme does not seem to compile

### ISSUE THREAD

**Issue by AndrejMitrovic** [aliak00/optional#59]
Hi,

I'm trying this out with DMD v2.102.0 with the `Includes a generic or range algorithm` example on https://github.com/aliak00/optional#features:

```D
import optional;

void main() {
    auto a = some(3);
    auto b = a.or(7);
}
```

```
lib/optional/source/optional/or.d(152,13): Error: expected return type of `int`, not `Optional!int`:
lib/optional/source/optional/or.d(150,13):        Return type of `int` inferred here.
lib/optional/source/optional/or.d(189,17): Error: template instance `optional.or.or!(Optional!int, int).or.or!(elseValue, Optional!int)` error instantiating
test.d(5,18):        instantiated from here: `or!(Optional!int, int)`
```

This works however:

```D
    auto a = some(3);
    auto b = a.or(some(7));
```

Maybe just `some()` was missing?

**Comment by SimonN** [comment]
Yes, `or(some(7))` works and has the wanted meaning. This should be fixed in the readme.

There is a deeper usability concern here. I remember I fell for this, too, and it wasn't immediately clear for me either how to fix the usercode. It would be nice if the error told you explicitly to use `or(some(7))`, or if the library accepted `or(7)` outright.

I believe that the optional library wants to be really explicit with these things. There should be only non-misleading auto-conversion, and there should be no accidental hijacking of methods (= an `optional!X` should not call or override methods of `X` by accident). This is why there is `oc` for dispatching to methods.

**Comment by SimonN** [comment]
The readme should use `a.frontOr(7)`.

The general rule is that `or` on an `Optional!X` chains with another `Optional!X` and `frontOr` chains with a bare `X`.
